core-web/libs/sdk/react/src/lib/components/DotEditableText/DotEditableText.tsx

```diff
@@ -0,0 +1,135 @@
+        }
+
+        const content = contentlet[fieldName] || '';
+        editorRef.current?.setContent(content, { format });
+        setContent(content);
+    }, [contentlet, fieldName, format]);
+
+    useEffect(() => {
+        const onMessage = ({ data }: MessageEvent) => {
+            const { name, payload } = data;
+            if (name !== 'COPY_CONTENTLET_INLINE_EDITING_SUCCESS') {
+                return;
+            }
+
+            const { oldInode, inode } = payload;
+            const currentInode = contentlet.inode;
+            const shouldFocus = currentInode === oldInode || currentInode === inode;
+
+            if (shouldFocus) {
+                editorRef.current?.focus();
+            }
+        };
+
+        window.addEventListener('message', onMessage);
+
+        return () => {
+            window.removeEventListener('message', onMessage);
+        };
+    }, [contentlet.inode]);
+
+    const onMouseDown = (event: MouseEvent) => {
+        const { onNumberOfPages = 1 } = contentlet;
+
+        if (onNumberOfPages <= 1 || editorRef.current?.hasFocus()) {
+            return;
+        }
+
+        const { inode, languageId: language } = contentlet;
+
+        event.stopPropagation();
+        event.preventDefault();
+
+        try {
+            postMessageToEditor({
+                action: CUSTOMER_ACTIONS.COPY_CONTENTLET_INLINE_EDITING,
+                payload: {
+                    dataset: {
+                        inode,
+                        language,
+                        fieldName
+                    }
+                }
+            });
+        } catch (error) {
+            console.error('Failed to post message to editor:', error);
+        }
+    };
+
+    const onFocusOut = () => {
+        const editedContent = editorRef.current?.getContent({ format: format }) || '';
+
+        if (!editorRef.current?.isDirty() || !didContentChange(editedContent)) {
+            return;
+        }
+
+        const { inode, languageId: langId } = contentlet;
+
+        try {
+            postMessageToEditor({
+                action: CUSTOMER_ACTIONS.UPDATE_CONTENTLET_INLINE_EDITING,
+                payload: {
+                    content: editedContent,
+                    dataset: {
+                        inode,
+                        langId,
+                        fieldName
+                    }
+                }
+            });
+        } catch (error) {
+            console.error('Failed to post message to editor:', error);
+        }
+    };
+
+    const didContentChange = (editedContent: string) => {
+        return content !== editedContent;
+    };
+
+    if (!isInsideEditor) {
+        // We can let the user pass the Child Component and create a root to get the HTML for the editor
+        return <div dangerouslySetInnerHTML={{ __html: content }} />;
+    }
+
+    return (
+        <Editor
+            tinymceScriptSrc={`${DotCmsClient.dotcmsUrl}/ext/tinymcev7/tinymce.min.js`}
```

Comment:
What happen with the instance issue we talked about? also... `DotCmsClient.dotcmsUrl` might have `/` which will result in a `//` here.

Reply:
I'll use a URL object

Reply:
Done!